src/box/sql/func.c

```diff
@@ -1118,6 +1118,61 @@ func_now(struct sql_context *ctx, int argc, const struct Mem *argv)
 	mem_set_datetime(ctx->pOut, &dt);
 }
 
+/**
+ * Implementation of the DATE_PART() function.
+ *
+ * Returns the requested information from a DATETIME value.
+ */
+static void
+func_date_part(struct sql_context *ctx, int argc, const struct Mem *argv)
+{
+	assert(argc == 2);
+	(void)argc;
+	const struct Mem *part = &argv[0];
+	const struct Mem *date = &argv[1];
+	if (mem_is_any_null(part, date))
+		return;
+	assert(mem_is_str(part) && mem_is_datetime(date));
+	const char *str = tt_cstr(part->z, part->n);
+	const struct datetime *dt = &date->u.dt;
+	if (strcasecmp("millennium", str) == 0)
+		return mem_set_int64(ctx->pOut, datetime_millennium(dt));
+	if (strcasecmp("century", str) == 0)
+		return mem_set_int64(ctx->pOut, datetime_century(dt));
+	if (strcasecmp("decade", str) == 0)
+		return mem_set_int64(ctx->pOut, datetime_decade(dt));
+	if (strcasecmp("year", str) == 0)
+		return mem_set_int64(ctx->pOut, datetime_year(dt));
+	if (strcasecmp("quarter", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_quarter(dt));
+	if (strcasecmp("month", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_month(dt));
+	if (strcasecmp("week", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_week(dt));
+	if (strcasecmp("day", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_day(dt));
+	if (strcasecmp("dow", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_dow(dt));
+	if (strcasecmp("doy", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_doy(dt));
+	if (strcasecmp("hour", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_hour(dt));
+	if (strcasecmp("minute", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_min(dt));
+	if (strcasecmp("second", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_sec(dt));
+	if (strcasecmp("milliseconds", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_milliseconds(dt));
+	if (strcasecmp("microseconds", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_microseconds(dt));
+	if (strcasecmp("nanoseconds", str) == 0)
+		return mem_set_uint(ctx->pOut, datetime_nsec(dt));
```

Comment:
Is it standard that all parts are single but these 3 are plural?

Reply:
There is no nothing below SECONDS in the standard. I fixed these three values.